src/components/services/Transmission.vue

```diff
@@ -0,0 +1,194 @@
+      rate || 0
+    ) + ` ${units[i]}/s`
+  );
+};
+
+export default {
+  name: "Transmission",
+  mixins: [service],
+  props: { item: Object },
+  components: { Generic },
+  data: () => ({ dl: null, ul: null, count: null, error: null }),
+  computed: {
+    downRate: function () {
+      return displayRate(this.dl);
+    },
+    upRate: function () {
+      return displayRate(this.ul);
+    },
+  },
+  created() {
+    const updateInterval = parseInt(this.item?.updateInterval, 10) || 0;
+    if (updateInterval > 0) {
+      this.interval = setInterval(() => this.fetchData(), updateInterval);
+    }
+    if (this.item?.sessionId) {
+      this.sessionId = this.item.sessionId;
+    }
+
+    this.fetchData();
+  },
+  unmounted() {
+    if (this.interval) {
+      clearInterval(this.interval);
+    }
+  },
+  methods: {
+    fetchData: async function () {
+      try {
+        const auth = {};
+        if (this.item?.username && this.item?.password) {
+          auth.user = this.item.username;
+          auth.password = this.item.password;
+        }
+        const {
+          arguments: { activeTorrentCount, downloadSpeed, uploadSpeed },
+        } = await this.transmissionFetch("session-stats", auth);
+
+        this.error = false;
+        this.ul = uploadSpeed;
+        this.dl = downloadSpeed;
+        this.count = activeTorrentCount;
+      } catch (e) {
+        this.error = true;
+        console.error(e);
+      }
+    },
+    /**
+     * @param {string} method
+     * @param {{user: string, password: string}?} auth
+     * @returns {Promise<{result: string, arguments: any}>}
+     */
+    transmissionFetch: async function (method, auth) {
+      /**
+       * @type {RequestInit}
+       */
+      const options = {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          method: method,
+        }),
+      };
+
+      if (this.proxy?.useCredentials) {
+        options.credentials = "include";
+      }
+
+      // Each item can override the credential settings
+      if (this.item.useCredentials !== undefined) {
+        options.credentials =
+          this.item.useCredentials === true ? "include" : "omit";
+      }
+
+      // Add the credentials if they are set
+      if (auth?.user && auth?.password) {
+        options.headers["Authorization"] = `Basic ${btoa(
+          `${auth.user}:${auth.password}`
+        )}`;
+      }
+
+      // Add the session id if it is set
+      if (this?.sessionId) {
+        options.headers["X-Transmission-Session-Id"] = this.sessionId;
+      }
+
+      let url;
+      if (this.item?.rpc) {
+        url = this.item.rpc;
+      } else {
+        url = this.endpoint;
+        if (url.endsWith("/")) {
+          url = url.slice(0, -1);
+        }
+        if (url.endsWith("/web")) {
+          url = url.slice(0, -4);
+          url += "/rpc";
+        }
+      }
```

Comment:
We don't need another config key for `/rpc` endpoint. the `endpoint` config key is made for this. It's supposed to be the url where to fetch the data. You just have to write in the documentation that `endpoint` should contain the rpc url.